### PR TITLE
Change to C

### PR DESCRIPTION
**Pull request by aaronsarkissian**

### CODE REVIEW

**Comment by fxfactorial** [comment]
This isn't ready yet.

**Comment by aaronsarkissian** [comment]
@fxfactorial So, I should remove everything, such as **C++ includes** and everything that is **not C**, right? Or is there another issue that I missed?

**Comment by fxfactorial** [comment]
The project must compile and created a usable binary before I can accept a PR. How can I accepted a PR that makes the project not even build?

**Comment by aaronsarkissian** [comment]
@fxfactorial I tested on both Ubuntu and Mac and it just worked fine. What is the problem? Or you mean after future modification it should be fine too?

File: shell.cpp [hunk 1]
```diff
@@ -1,12 +1,9 @@
 #include "shell.h"
 
-using std::vector;
-using std::string;
-using std::cout;
+//using std::vector;
+//using std::string;
 using std::cin;
```
Comment (fxfactorial) [inline shell.cpp]: This is not C

File: shell.cpp [hunk 2]
```diff
@@ -90,66 +87,71 @@ int special_chars(char *unique) {
...
-  string input_line;
   char line[1024];
   char *argv[1024];
 
-  cout << "[oh my gosh shell] $ ";
-  getline(cin, input_line);
-  strcpy(line, input_line.c_str()); //string to char array converter
-  cout << give_me_color(1) << line << give_me_color(0) << endl;
+  printf("%s", "[oh my gosh shell] $ ");
+  cin.getline(line, 1024);
```
Comment (fxfactorial) [inline shell.cpp]: not C

**Comment by fxfactorial** [comment]
This is not a C project yet, still has C++ things in it.

**Comment by aaronsarkissian** [comment]
@fxfactorial Ok, I get it. I'll fix it soon.

File: shell.cpp [hunk 3]
```diff
@@ -122,7 +96,7 @@ void get_comand(void) {
   char *argv[1024];
 
   printf("%s", "[oh my gosh shell] $ ");
-  cin.getline(line, 1024);
+  fgets(line, 1024, stdin);
```
Comment (fxfactorial) [inline shell.cpp]: no, fgets is bad. You can still use `getline` in C.

File: shell.cpp [hunk 4]
```diff
@@ -92,11 +92,17 @@ void fork_exec(char *argv[]) {
 
 void get_comand(void) {
 
-  char line[1024];
-  char *argv[1024];
+  size_t bufsize = 1024;
+  char *line;
+  char *argv[bufsize];
+  line = (char *)malloc(bufsize * sizeof(char));
```
Comment (fxfactorial) [inline shell.cpp]: No casting result of malloc in C

Comment (fxfactorial) [inline shell.cpp]: Plus no need to do sizeof(char), the C standard says that char is always 1 byte.

**Pull request merged**